core/src/main/java/org/apache/calcite/sql/dialect/SparkSqlDialect.java

```diff
@@ -636,6 +640,23 @@ private SqlCharStringLiteral creteDateTimeFormatSqlCharLiteral(String format) {
       String standardDateFormat, Map<SqlDateTimeFormat, String> dateTimeFormatMap) {
     return super.getDateTimeFormatString(standardDateFormat, dateTimeFormatMap);
   }
+
+  private void unparsePlusMinus(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    switch (call.getOperator().getName()) {
+    case "TIMESTAMP_ADD":
+      call.operand(0).unparse(writer, leftPrec, rightPrec);
+      writer.print("+ ");
```

Comment:
create a separate method for this and pass `sign` "+" or "-" as argument to that method.